src/fhir/api.test.ts

```diff
@@ -1,103 +1,17 @@
+import { sendBundle, deletePatients, getPatients, getObservations } from "./testUtilities";
 
 const fhir = new Fhir();
 
 const reportedGenes = [geneCoding("HGNC:4389", "GNA01")];
-const FHIR_URL = process.env.REACT_APP_FHIR_URL || "";
-
-const checkResponseOK = async (response: Response) => {
-  const r = await response.json();
-
-  if (!response.ok) {
-    console.error(r.body);
-    throw new Error(response.statusText);
-  }
-  if (r.type !== "bundle-response") {
-    return r;
-  }
-
-  const errors = (r as BundleResponse).entry
-    .filter((entry) => entry.response.status.toString().startsWith("4"))
-    .map((entry) => entry.response.outcome.issue);
-
-  if (errors.length > 1) {
-    errors.forEach((issue) => {
-      let message = "unknown error in bundle";
-      if (issue && issue.length > 0) {
-        message = issue[0].diagnostics;
-      }
-      throw new Error(message);
-    });
-  }
-
-  console.debug(`check response: ${JSON.stringify(r, null, 2)}`);
-  return r;
-};
-
-const getPatients = async (identifier?: string): Promise<Bundle> => {
-  let url = `${FHIR_URL}/Patient`;
-  if (identifier) url = `${FHIR_URL}/Patient?identifier=${identifier}`;
-  const response = await fetch(url);
-  return await checkResponseOK(response);
-};
-
-const getObservations = async (identifier?: string): Promise<Bundle> => {
-  let url = `${FHIR_URL}/Observation`;
-  if (identifier) url = `${FHIR_URL}/Observation?identifier=${identifier}`;
-  const response = await fetch(url);
-  return await checkResponseOK(response);
-};
-
-const sendBundle = async (bundle: Bundle) => {
-  const sentBundle = await fetch(`${FHIR_URL}/`, {
-    method: "POST",
-    body: JSON.stringify(bundle),
-    headers: {
-      "Content-Type": "application/json",
-    },
-  });
-  await new Promise((r) => setTimeout(r, 1500));
-  return sentBundle;
-};
-
-/**
- * Given an ID string
- * this will delete the chosen patient's records, otherwise it will remove all patients by looping
- * through and extracting IDs from the search bundle
- */
-const deletePatients = async (patientId?: string) => {
-  const patientData = await getPatients();
-  if (!("entry" in patientData)) {
-    console.debug("Nothing to delete; no patients in database");
-    return;
-  }
-  if (patientId) {
-    await deleteAndCascadeDelete([patientId]);
-  } else {
-    const patientIds = patientData.entry?.map((entry) => entry.resource?.id) as string[];
-    await deleteAndCascadeDelete(patientIds);
-  }
-
-  // await new Promise((r) => setTimeout(r, 1500));
-};
-
-const deleteAndCascadeDelete = async (patientIds: string[]) => {
-  console.debug(`deleting ids: ${patientIds}`);
-  for (const patientId of patientIds) {
-    const response = await fetch(`${FHIR_URL}/Patient/${patientId}?_cascade=delete`, {
-      method: "DELETE",
-    });
-    await checkResponseOK(response);
-  }
-};
+export const FHIR_URL = process.env.REACT_APP_FHIR_URL || "";
```

Comment:
this should probably be in testUtilities, and then we use it in all the methods there so we don't have to export it